src/2_storage.js

```diff
@@ -136,39 +136,41 @@ BranchStorage.prototype['session'] = function() {
 	return webStorage(false);
 };
 
-var cookies = function(perm) {
+var cookies = function() {
 	var setCookie = function(key, value) {
-		var expires = '';
-		if (perm) {
-			var date = new Date();
-			date.setTime(date.getTime() + COOKIE_MS);
-			expires = '; branch_expiration_date=' + date.toGMTString() +
-				'; expires=' + date.toGMTString();
+		document.cookie = key + '=' + value + '; path=/';
+	};
+	var removeCookie = function(key, ignorePrefix) {
+		var expires = 'Thu, 01 Jan 1970 00:00:01 GMT';
+		if (ignorePrefix) {
+			document.cookie = key + '=; expires=' + expires + '; path=/';
+		}
+		else {
+			document.cookie = prefix(key) + '=; expires=' + expires + '; path=/';
 		}
-		document.cookie = key + '=' + value + expires + '; path=/';
 	};
 	return {
 		getAll: function() {
-			var cookieArray = document.cookie.split(';');
 			var returnCookieObject = { };
+			var cookieArray = document.cookie.split(';');
 			for (var i = 0; i < cookieArray.length; i++) {
-				var cookie = cookieArray[i].replace(' ', '');
-				cookie = cookie.substring(0, cookie.length);
-				if (cookie.indexOf(BRANCH_KEY_PREFIX) !== -1) {
-					var splitCookie = cookie.split('=');
-					returnCookieObject[trimPrefix(splitCookie[0])] = retrieveValue(splitCookie[1]);
-				}
+				var cookie = cookieArray[i].trim();
+				var firstEqualSign = cookie.indexOf("=");
+				var cookieName = trimPrefix(cookie.substring(0, firstEqualSign));
+				var cookieValue = cookie.substring(firstEqualSign + 1, cookie.length);
+				returnCookieObject[cookieName] = cookieValue;
 			}
 			return returnCookieObject;
 		},
 		get: function(key) {
-			var keyEQ = prefix(key) + '=';
+			key = prefix(key);
 			var cookieArray = document.cookie.split(';');
 			for (var i = 0; i < cookieArray.length; i++) {
-				var cookie = cookieArray[i];
-				cookie = cookie.substring(1, cookie.length);
-				if (cookie.indexOf(keyEQ) === 0) {
-					return retrieveValue(cookie.substring(keyEQ.length, cookie.length));
```

Comment:
the `retrieveValue` function will coerce a boolean. Are you sure we no longer need that? If we do a check for false and the value is "false", it may cause some unexpected behavior

Reply:
I've added this check back in:

https://github.com/BranchMetrics/web-branch-deep-linking/pull/521/files#diff-3351e3fcc07fe61dec4286479ff4455dR70

Anytime we want to send the value of a cookie back, the retrieveValue function will ensure that boolean values are properly returned.